### PR TITLE
Implemented Packaging of Morganey Source Files

### PR DESCRIPTION
**Pull request by 06needhamt**
Reproduction steps

```
-     $ gradle build
-     $ gradle publish
-     $ cd plugin-test
-     $ gradle build
-     $ unzip -l ./build/libs/plugin-test-0.1.jar
```

Observed

```
Archive:  ./build/libs/plugin-test-0.1.jar
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  10-05-2016 14:57   META-INF/
       25  10-05-2016 14:57   META-INF/MANIFEST.MF
---------                     -------
       25                     2 files
```

Expected

Archive should include `src/test.mng` and `src/other.mng`

### CODE REVIEW

**Review by rexim — CHANGES_REQUESTED** [review]
Please squash the commits as you suggested in https://github.com/morganey-lang/morganey-gradle-plugin/pull/7#issuecomment-251667382

**Comment by 06needhamt** [comment]
That is done upon merging the PR so i will do it when we merge it

**Comment by rexim** [comment]
> That is done upon merging the PR so i will do it when we merge it
1. That is NOT true. Please read git documentation on how to squash commits. If you don't know how to use git, just ask someone to squash your branch.
2. You won't be able to squash commits after the merge because push to the master is forbidden
3. Even if you could it is not possible to squash commit in the master without force pushing the changes afterwards. And foce pushing to master is not acceptible in projects developed by several people.

I'm not merging and not even reviewing that until the branch is squashed. I'm sorry.

**Comment by ForNeVeR** [comment]
> That is NOT true.

You're not completely right: maintainer can squash commit right on GitHub upon merge. But I don't know anyone who does that. So, usually that's a PR author responsibility.

**Comment by 06needhamt** [comment]
I'm Pretty sure the option is on the little arrow after Merge Pull Request let me check

**Comment by 06needhamt** [comment]
See 
squash

**Comment by rexim** [comment]
@06needhamt squash AND merge. We need "squash then review and only then maybe merge". I don't see that option there.

**Comment by rexim** [comment]
And I'm pretty sure documentation about squashing commits is here https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

**Comment by 06needhamt** [comment]
@rexim ah ok i will do it like this

**Comment by 06needhamt** [comment]
@rexim @ForNeVeR Do one of you mind doing it i can't get it to work i'm really sorry

**Comment by rexim** [comment]
@06needhamt sure, I can do that

**Comment by rexim** [comment]
Done. Hope I didn't break anything.

**Comment by ForNeVeR** [comment]
@06needhamt there's nothing wrong to ask for help!

**Comment by rexim** [comment]
@06needhamt thanks for the PR. But I'm gonna review it tomorrow sorry. I have lots of stuff planned for today.

**Comment by 06needhamt** [comment]
@rexim OK sorry for the mess

**Comment by ForNeVeR** [comment]
Hey, guys, I'm confused. Is it still work in progress, or do I need to review that already? If it's in progress and @06needhamt doesn't have any particular problem at the moment, then maybe this should be assigned to him?

**Comment by 06needhamt** [comment]
@ForNeVeR  This is a work in progress and there is a problem which is outlined in the first comment of the PR

**Comment by rexim** [comment]
@06needhamt does this problem exist in the master branch?

**Comment by 06needhamt** [comment]
@rexim nope this issue is not present in the master branch

**Comment by rexim** [comment]
That means there is no particular problem and this is work in progress.

**Comment by rexim** [comment]
@06needhamt please ask for the review once #2 is done. Thanks!

**Comment by 06needhamt** [comment]
The feature mentioned by #2 is now implemented. Please review whenever you can

**Comment by rexim** [comment]
@06needhamt please do not create additional labels. Labels are part of the development process. And development process is organized by the Project Leader.

**Comment by 06needhamt** [comment]
@rexim You should mention that in the guidelines

**Comment by rexim** [comment]
@06needhamt to indicate that the PR is ready for review just remove `[WIP]` from the title and assign the PR accordingly.

**Comment by rexim** [comment]
> @rexim You should mention that in the guidelines

Yep, just made a note for myself. :)

**Comment by rexim** [comment]
@06needhamt we are pretty young community and not everything is covered by rules and guidelines yet. But we are working on that. :)

**Comment by ForNeVeR** [comment]
@06needhamt the task stated in the first post isn't still achieved. I've tried to run this build, and it creates **two** files:
- `build/libs/Morganey-Files-0.1.jar` with `other.mng` and `test.mng`
- `build/libs/plugin-test-0.1.jar` without any `*.mgn` files

Is that intentional? I thought they all should be in `plugin-test-0.1.jar`.

File: src/main/groovy/me/rexim/morganey/gradle/MorganeyPlugin.groovy
Comment (ForNeVeR) [inline src/main/groovy/me/rexim/morganey/gradle/MorganeyPlugin.groovy]: Aha, it seems that the plugin uses this name instead of the one I think it should (`project-test.jar`). I wonder, what will happen if we remove this line?

Comment (06needhamt) [reply src/main/groovy/me/rexim/morganey/gradle/MorganeyPlugin.groovy]: @ForNeVeR That fixes the issue

Comment (ForNeVeR) [reply src/main/groovy/me/rexim/morganey/gradle/MorganeyPlugin.groovy]: Ok cool! Will you commit this? I'm ready to merge then.

Comment (06needhamt) [reply src/main/groovy/me/rexim/morganey/gradle/MorganeyPlugin.groovy]: Its Committed Sorry I had to leave suddenly

Comment (ForNeVeR) [reply src/main/groovy/me/rexim/morganey/gradle/MorganeyPlugin.groovy]: Not a problem at all! Thank you.

**Comment by ForNeVeR** [comment]
@rexim could you please review again? It's now my responsibility, but you left your negative review result there, and I'd say it's blocking me from merging this.

I'll review and clean this up tonight.

**Comment by ForNeVeR** [comment]
@06needhamt please merge `master` into your branch.

**Review by rexim — APPROVED** [review]

File: src/main/groovy/me/rexim/morganey/gradle/MorganeyPlugin.groovy
Comment (ForNeVeR) [inline src/main/groovy/me/rexim/morganey/gradle/MorganeyPlugin.groovy:20]: In my issue definition, I meant `package` as a generic module name. It could as well be `me.rexim` or `me.fornever`, and should anyways create the corresponding directory structure inside of the resulting artifact.

I suggest to remove `/package` part here. Will it work without it?

**Review by ForNeVeR — CHANGES_REQUESTED** [review]

**Comment by 06needhamt** [comment]
I'm out at the moment I will take a look when I get home

**Comment by ForNeVeR** [comment]
Not a problem. You know what? Let me check it myself. I should also contribute here after all, right? :)

**Comment by ForNeVeR** [comment]
@06needhamt thank you!

**Pull request merged**
